clients/src/main/java/org/apache/kafka/clients/consumer/OffsetAndTimestamp.java

```diff
@@ -20,14 +20,13 @@
 
 /**
  * A container class for offset and timestamp.
- *
- * Offset must be non-negative.
- *
- * The timestamp should never be negative, unless it is invalid.  This could happen when
- * handling a response from a broker that doesn't support KIP-79.
  */
 public final class OffsetAndTimestamp {
+    // The timestamp should never be negative, unless it is invalid.  This could happen when
```

Comment:
This comment still makes me uncomfortable. For a public-facing classes, it would be nice to have surer guarantees about the class invariants (e.g. by checking that the timestamp is not negative in the constructor). I wonder if there is a different way we can represent the sentinel value internally in `Fetcher.handleListOffsetResponse`. For example maybe we just need a static instance that we can use in `Fetcher` whenever we need the sentinel and we can use `==` to check for it?